src/Illuminate/Database/Query/Builder.php

```diff
@@ -1995,15 +1989,56 @@ protected function enforceOrderBy()
      */
     public function pluck($column, $key = null)
     {
-        $results = $this->get(is_null($key) ? [$column] : [$column, $key]);
+        $queryResult = $this->onceWithColumns(
+            is_null($key) ? [$column] : [$column, $key],
+            function () {
+                return $this->processor->processSelect($this, $this->runSelect());
+            }
+        );
 
         // If the columns are qualified with a table or have an alias, we cannot use
         // those directly in the "pluck" operations since the results from the DB
         // are only keyed by the column itself. We'll strip the table out here.
-        return $results->pluck(
-            $this->stripTableForPluck($column),
-            $this->stripTableForPluck($key)
-        );
+        $column = $this->stripTableForPluck($column);
+        $key = $this->stripTableForPluck($key);
+
+        $results = [];
+
+        foreach ($queryResult as $row) {
+            $itemValue = $row->$column;
+
+            if (is_null($key)) {
```

Comment:
I would move the `is_null($key)` function call outside of the loop.

(replacing with `$key === null` would be fine too, but you know the laravel convention...)